doobie/src/main/scala/za/co/absa/fadb/doobie/DoobieStreamingEngine.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.fadb.doobie
+
+import cats.effect.Async
+import doobie.Transactor
+import doobie.implicits.toDoobieStreamOps
+import doobie.util.Read
+import za.co.absa.fadb.DBStreamingEngine
+
+import scala.language.higherKinds
+
+/**
+ *  `DoobieStreamingEngine` is a class that represents a database engine.
+ *  It provides methods to execute streaming queries from a database.
+ *  @tparam F - The type of the context in which the database queries are executed.
+ */
+class DoobieStreamingEngine[F[_]: Async](val transactor: Transactor[F], defaultChunkSize: Int = 512)
+    extends DBStreamingEngine[F] {
+
+  /** The type of Doobie queries that produce `R` */
+  type QueryType[R] = DoobieQuery[R]
+
+  /**
+   *  Executes a Doobie query and returns the result as an `fs2.Stream[F, R]`.
+   *
+   *  @param query the Doobie query to execute
+   *  @param readR the `Read[R]` instance used to read the query result into `R`
```

Comment:
this is not a separate parameter, but if you want to keep it you can modify this (here and below) somehow so that it's all under `query` param docs

Reply:
Thanks, didn't notice it. Removed it from methods where the read instance is not one of parameters.